exec/java-exec/src/main/resources/rest/profile/list.ftl

```diff
@@ -37,7 +37,15 @@
       <strong>No running queries.</strong>
     </div>
   </#if>
-  <h3>Completed Queries</h3>
+  <table width="100%">
+    <tr>
+      <td><h3>Completed Queries</h3></td>
+      <td align="right">
+        <form action="/profiles" method="get"><span title="Max number of profiles to list">Show </span>
+        <input id="fetchMax" type="text" size="7" name="max" value="${model.getFinishedQueries()?size}">
```

Comment:
Better! But, the number to show is the *target* number of profiles, not the *actual* number. If we show the actual number, then we get this:

* Show profiles with the target of 100.
* We actually have 10 profiles, so that is displayed in the "max" control.
* Two more profiles come in, now we have 12.
* Hit the "Go" button. Only 10 profiles appear because that was the prior count.
* Must manually type a larger number to see more.

What we want:

* Show profiles with the target of 100.
* We actually have 10 profiles, but the target is 100, so 100 is displayed in the "max" control.
* Two more profiles come in, now we have 12.
* Hit the "Go" button. All 12 profiles appear.

To make this work:

* Add to the model the *target* number: either the default of 100 or the number received via the max parameter.
* Display that model number in the "max" control.